tests/test_linkextractors.py

```diff
@@ -851,3 +852,36 @@ def test_skip_bad_links(self):
                 ),
             ],
         )
+
+    def test_link_allowed_is_false_with_empty_url(self):
+        mock_link = Mock()
+        mock_link.url = ""
+        expected = False
+
+        actual = LxmlLinkExtractor()._link_allowed(mock_link)
+
+        self.assertEqual(expected, actual)
+
+    def test_link_allowed_is_false_with_bad_url_prefix(self):
+        mock_link = Mock()
+        mock_link.url = "htp://should_be_http.com"
+        expected = False
+
+        actual = LxmlLinkExtractor()._link_allowed(mock_link)
+
+        self.assertEqual(expected, actual)
+
+    def test_link_allowed_is_false_with_missing_url_prefix(self):
+        mock_link = Mock()
+        mock_link.url = "should_have_prefix.com"
+        expected = False
+
+        actual = LxmlLinkExtractor()._link_allowed(mock_link)
+
+        self.assertEqual(expected, actual)
```

Comment:
:lipstick: This is a very personal opinion, feel free to ignore it, but I feel like a more compact version of these tests would still be readable:

```suggestion
    def test_link_allowed_is_false_with_missing_url_prefix(self):
        link = Link("should_have_prefix.com")
        self.assertFalse(LxmlLinkExtractor()._link_allowed(link))
```